include/tiramisu/core.h

```diff
@@ -1234,9 +1238,10 @@ class function
     /**
      * Align schedules dimensions and adds the computation's order to them. 
      * This is done to correctly invoke calculate_dep_flow() method that performs dependence analysis
-     * It calls gen_ordering_schedules() and align_schedules() function's methods internally
+     * It calls gen_ordering_schedules() and align_schedules() function's methods internally.
+     * if \p reset_static_dimesion is true then it will also call function.reset_all_static_dims_to_zero() before ordering.
     */
-    void prepare_schedules_for_legality_checks() ;
+    void prepare_schedules_for_legality_checks(bool reset_static_dimesion = false) ;
 
```

Comment:
if set to true, it will reset the static beta dimension then rewrite them from the scheduling graph sched_graph using function->gen_ordering_schedules() ...
the default behavior is to align then order the schedules.
the alternative behavior is to align, reset beta dimension, then ordering the schedules.